met-web/tests/unit/components/widgets/VideoWidget.test.tsx

```diff
@@ -0,0 +1,151 @@
+import { render, waitFor, screen, fireEvent } from '@testing-library/react';
+import React from 'react';
+import '@testing-library/jest-dom';
+import EngagementForm from '../../../../src/components/engagement/form';
+import { setupEnv } from '../setEnvVars';
+import * as reactRedux from 'react-redux';
+import * as reactRouter from 'react-router';
+import * as engagementService from 'services/engagementService';
+import * as widgetService from 'services/widgetService';
+import * as engagementMetadataService from 'services/engagementMetadataService';
+import * as membershipService from 'services/membershipService';
+import * as engagementSettingService from 'services/engagementSettingService';
+import * as VideoService from 'services/widgetService/VideoService';
+import { Box } from '@mui/material';
+import { WidgetType } from 'models/widget';
+import { draftEngagement, mockVideo, videoWidget, engagementMetadata } from '../factory';
+import { USER_ROLES } from 'services/userService/constants';
+import { EngagementSettings, createDefaultEngagementSettings } from 'models/engagement';
+
+jest.mock('components/map', () => () => <div></div>);
+
+const mockEngagementSettings: EngagementSettings = {
+    ...createDefaultEngagementSettings(),
+};
+
+jest.mock('axios');
+jest.mock('react-redux', () => ({
```

Comment:
I wonder if we could set up a module that gets included for these widget tests. I noticed there's a lot of the same code declared at the top of each

Reply:
Moved most of the Jest mock and Spy codes to a outside reusable functions